File: tests/fixtures/sequelize_app/service_b.ts
import { User, Order, LineItem } from "./models";

export async function listOrders(userId: number) {
  const u = await User.findByPk(userId);
  const orders = await Order.findAll({ where: { userId: u!.id } });
  const items = await LineItem.findAll({
    where: { orderId: orders.map((o) => o.id) },
  });
  return { u, orders, items };
}

File: tests/fixtures/sqlalchemy_app/service_b.py
"""Service B — different code, same JOIN. Should cluster with service_a."""

from .models import LineItem, Order, User


def list_user_orders(user_id, since=None):
    rows = select(User, Order, LineItem).where(Order.user_id == User.id)
    if since:
        rows = rows.where(Order.created_at > since)
    return rows


def lookup_unrelated():
    return select(User).where(User.id == 1)


def select(*args):
    return _Q()


class _Q:
    def where(self, *_):
        return self
